### DISCUSSION TITLE
Unable to obtain PageCount for PDFs

### DISCUSSION THREAD

**Discussion by mirko-horvacki** [dlemstra/Magick.NET#1359] [Help]
Hi,

I tried to obtain Page Count using following code:

`var info = PdfInfo.Create(fileName);`
`isMultipage = info.PageCount > 1;`

Code works perfectly when run locally on my machine but - it does not work when deployed to Azure App Service.

GhostScript is available on machine where my service is deployed, since I am using already Image Magick for different cases (one of them is rasterization which utilizes GhostScript) and those cases are fine.

Here is what exception from logs says:

FailedToExecuteCommand `"gswin64c.exe" -q -dQUIET -dSAFER -dBATCH -dNOPAUSE -dNOPROMPT --permit-file-read="7a3ea60f-8e28-48d3-8d8d-1a2841d200c5.pdf" -sPDFPassword="" -c "(7a3ea60f-8e28-48d3-8d8d-1a2841d200c5.pdf) (r) file runpdfbegin pdfpagecount = quit"' (-1073741819) @ error/delegate.c/ExternalDelegateCommand/521

What does error/delegate.c/ExternalDelegateCommand/521 mean?

Do you know maybe how I could know why command failed to execute?

It does not matter which PDF file I use, for all - exception is the same.

Thanks in advance for help!

Kind regards,

Mirko

**Comment by dlemstra** [comment]
It suspect that `gswin64c.exe` is not in your `%PATH%`. I am not sure why ImageMagick can find it for other uses cases but that probably has to do with it using a library call instead of using the executable directly.

**Reply by mirko-horvacki** [reply]
Hm, I tried to set %PATH%, using this lines:

`var ghostScriptPath = Path.GetDirectoryName(Assembly.GetEntryAssembly().Location);`
`MagickNET.SetGhostscriptDirectory(ghostScriptPath);`

Now the error is slightly different:

FailedToExecuteCommand `"C:\home\site\wwwroot\gswin64c.exe" -q -dQUIET -dSAFER -dBATCH -dNOPAUSE -dNOPROMPT --permit-file-read="eb619828-4bef-4310-8e83-9c3fb4f2b31e.pdf" -sPDFPassword="" -c "(eb619828-4bef-4310-8e83-9c3fb4f2b31e.pdf) (r) file runpdfbegin pdfpagecount = quit"' (-1073741819) @ error/delegate.c/ExternalDelegateCommand/521

New is only that now full path of exe file is used for command. And exe file is located at this full path, I checked via Kudu Diagnostic Console.

What could be still wrong? Do you know maybe what these numbers (1073741819 and 521) mean?

Thanks in advance!

Kind regards,

Mirko

**Reply by dlemstra** [reply]
The number `521` is a line number in the code of ImageMagick in the `delegate.c` file. And `-1073741819` is the exit code of the program. It might be possible that the executable is unable to find it's dependencies. Or maybe your process cannot execute a file in the root of the project? What happens when you move the ghostscript files to the bin folder?
